### PR TITLE
Feature/te 297 launch tests from tsl

### PR DESCRIPTION
**Pull request by kpanier**

### CODE REVIEW

File: rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend [hunk 1]
```diff
@@ -140,18 +168,24 @@ class TclLauncherUi implements Launcher {
 		if (selection.size > 1) {
 			return selection.toList.map[testCaseListFromSelection].flatten.toList
 		} else {
-			if (selection.firstElement instanceof IFolder) {
-				return selection.firstElement.testCaseListFromSelection.toList
-			}
-			return #[launchShortcutUtil.getQualifiedNameForTestInTcl(selection.firstElement as IResource).toString]
+			return selection.firstElement.testCaseListFromSelection.toList
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: What about the case that multiple elements are selected? Why is this removed in this change?

File: rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend [hunk 2]
```diff
@@ -140,18 +168,24 @@ class TclLauncherUi implements Launcher {
...
 		}
 	}
 
 	def Iterable<String> getTestCaseListFromSelection(Object sel) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: From the name of the method I would expect that it operates on an `IStructuredSelection`. Also the parameter name suggests that. Can we pass in the whole selection here and let the method deal with it instead of the caller?

File: rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend [hunk 3]
```diff
@@ -140,18 +168,24 @@ class TclLauncherUi implements Launcher {
...
 		if (sel instanceof IFolder) {
 			sel.testCasesFromFolder
 		} else {
-			#[launchShortcutUtil.getQualifiedNameForTestInTcl(sel as IResource).toString]
+			if (sel instanceof IResource) {
+				if (sel.fileExtension.equalsIgnoreCase("tsl")) {
+					val uri = URI.createPlatformResourceURI(sel.locationURI.toString, true).deresolve(
+						URI.createPlatformResourceURI(sel.project.locationURI.toString, true))
+					val secondURI = URI.createPlatformResourceURI(uri.toString, true)
+					tslIndex.get(secondURI).map[toString]
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: Please use an explicit `return` statement

File: rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend [hunk 4]
```diff
@@ -140,18 +168,24 @@ class TclLauncherUi implements Launcher {
...
+				} else {
+					#[launchShortcutUtil.getQualifiedNameForTestInTcl(sel).toString]
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: Please use an explicit `return` statement

File: rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend [hunk 5]
```diff
@@ -74,6 +84,24 @@ class TclLauncherUi implements Launcher {
 		return false
 	}
 
+	def createTestCaseIndex(IProject project) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: public methods should have explicit return types

Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: Is the parameter `project` used in this method? Couldn't find it.

Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: I think we don't need this method at all. When we need a resource to a given URI we can use `IResourceDescriptions.getResourceDescription(URI)`. No need to fill a custom map beforehand.

File: rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend (hunk 1 above)
Comment (kpanier) [inline rcp/org.testeditor.rcp4.views.tcltestrun/src/org/testeditor/rcp4/views/tcltestrun/TclLauncherUi.xtend]: The if statement checks the selection size. The logic was dublicated in the further version the function handles the multiple selection.
